src/inmanta/module.py

```diff
@@ -501,30 +501,33 @@ def log_pre_install_information(self, module_name: str, module_spec: List[Inmant
         raise NotImplementedError("Abstract method")
 
     def _log_version_snapshot(self, header: Optional[str], version_snapshot: Dict[str, "Version"]) -> None:
+        if header:
+            LOGGER.debug(header)
         if version_snapshot:
-            out = [header] if header is not None else []
-            out.extend(f"{mod}: {version}" for mod, version in version_snapshot.items())
-            LOGGER.debug("\n".join(out))
+            for mod, v in version_snapshot.items():
+                LOGGER.debug(f"{mod}: {v}")
 
     def _log_snapshot_difference(
         self, version_snapshot: Dict[str, "Version"], previous_snapshot: Dict[str, "Version"], header: Optional[str]
     ) -> None:
-        set_pre_install = set(previous_snapshot.items())
-        set_post_install = set(version_snapshot.items())
-        updates_and_additions = set_post_install - set_pre_install
-
+        set_pre_install: Set[Tuple[str, "Version"]] = set(previous_snapshot.items())
+        set_post_install: Set[Tuple[str, "Version"]] = set(version_snapshot.items())
+        updates_and_additions: Set[Tuple[str, "Version"]] = set_post_install - set_pre_install
+        kept: Set[Tuple[str, "Version"]] = set_pre_install.intersection(set_post_install)
+
+        if header:
+            LOGGER.debug(header)
+        for inmanta_module_name, package_version in sorted(kept):
+            LOGGER.debug(inmanta_module_name + ": " + str(package_version))
```

Comment:
I could be wrong, but I think it was suggested at some point to only log a "real" diff view (aka only things that got actually changed)
```suggestion

```